projects/core/src/cart/store/reducers/multi-cart.reducer.ts

```diff
@@ -28,7 +28,7 @@ export function activeCartReducer(
       }
       return state;
     case CartActions.CLEAR_CART_STATE:
-      return activeCartInitialState;
+      return state === null ? null : activeCartInitialState;
```

Comment:
Just curious... Can the value ever be undefined?

Reply:
1. Default value for cart is `''`. However when we use state persistence for cart the default is changed with meta reducer to `null` and the value stays `null` until the `CartStatePersistenceService` initializes it with the value from the localStorage or with `''` if it is not present in storage. You can tread `null` as uninitialized state.
2. Not in basic spartacus.